src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.DataConverter;
+import core.basesyntax.validator.DataValidator;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    private static final String COMMA = ",";
+    private final DataValidator validator;
+
+    public DataConverterImpl(DataValidator validator) {
+        this.validator = validator;
+    }
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputData) {
+        validator.validate(inputData);
+        List<FruitTransaction> fruitTransactions = new ArrayList<>();
+        for (String line : inputData.subList(1, inputData.size())) {
+            String[] partsOfData = line.split(COMMA);
+            if (partsOfData.length != 3) {
+                throw new IllegalArgumentException("Incorrect input data " + line);
+            }
+
+            try {
+                Operation fruitOperation = getOperationByCode(partsOfData[0]);
+                String fruit = partsOfData[1];
+                int quantity = Integer.parseInt(partsOfData[2]);
+                fruitTransactions.add(new FruitTransaction(fruitOperation, fruit, quantity));
+            } catch (IllegalArgumentException e) {
+                throw new IllegalArgumentException("Invalid data " + line
+                        + " - " + e.getMessage());
```

Comment:
While catching `IllegalArgumentException` is useful, consider logging the error or providing more detailed information to help trace the source of the problem.